File: ccif/src/java/other/org/himalay/ccif/codeVersionTableMsg.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class codeVersionTableMsg extends BinStruct implements PublicBinMsg {




    // members variables
    // purchaseType
    public short purchaseType ;
    // purchasePrice
    public int purchasePrice ;
    // videoFormat
    public DescriptionData videoFormat ;

    public codeVersionTableMsg () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize purchaseType

        // Initialize purchasePrice

        // Initialize videoFormat
        videoFormat = new DescriptionData();
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read purchaseType
        {
            purchaseType=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read purchasePrice
        {
            purchasePrice=istream.readUnsignedShort();
            retVal+=2;
        }
        // read videoFormat
        retVal += videoFormat.read(istream);

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;



        {   /** fix dependent sizes for videoFormat **/
        }

        // write purchaseType
        ostream.writeByte(purchaseType);
        retVal +=1;
        // write purchasePrice
        ostream.writeShort(purchasePrice);
        retVal +=2;
        // write videoFormat
        if (videoFormat!=null)retVal +=videoFormat.write(ostream);
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("codeVersionTableMsg\n");
        dc.increaseIndent();
        int retVal= 0;
        // write purchaseType
        dc.indent();
        dc.getPs().println("purchaseType="+purchaseType+"(0x"+ Integer.toHexString(purchaseType) + ")" );
        // write purchasePrice
        dc.indent();
        dc.getPs().println("purchasePrice="+purchasePrice+"(0x"+ Integer.toHexString(purchasePrice) + ")" );
        // write videoFormat
        if ( videoFormat != null ) {
            dc.indent();
            dc.getPs().println("videoFormat") ;
            retVal +=videoFormat.dump(dc);
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for purchaseType
    //public short getPurchaseType()
    //{
    //    return purchaseType ;
    //}


    // Setter for purchaseType
    //public void setPurchaseType(short val)
    //{
    //    this.purchaseType= val;
    //}
    // Getter for purchasePrice
    //public int getPurchasePrice()
    //{
    //    return purchasePrice ;
    //}


    // Setter for purchasePrice
    //public void setPurchasePrice(int val)
    //{
    //    this.purchasePrice= val;
    //}
    // Getter for videoFormat
    //public DescriptionData getVideoFormat()
    //{
    //    return videoFormat ;
    //}


    // Setter for videoFormat
    //public void setVideoFormat(DescriptionData val)
    //{
    //    this.videoFormat= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/codeVersionTableMsg.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:13 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "codeVersionTableMsg.h"

#ifdef org_himalay_ccif_codeVersionTableMsg__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
codeVersionTableMsg::codeVersionTableMsg() {
    references= 0;


    // purchaseType
    purchaseType= (ui8)0;
    // purchasePrice
    purchasePrice= (ui16)0;
    // videoFormat
    org_himalay_ccif_DescriptionData__NEW( videoFormat);
}
// Destructor
codeVersionTableMsg::~codeVersionTableMsg() {
    // purchaseType

    // purchasePrice

    // videoFormat
    org_himalay_ccif_DescriptionData__DELETE( videoFormat);
}

int codeVersionTableMsg::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // purchaseType
    {
        purchaseType=(istream.readUI8());
        retVal+=1;
    }
    // purchasePrice
    {
        purchasePrice=istream.readUI16();
        retVal+=2;
    }
    // videoFormat
    retVal += videoFormat->read(istream);
    return retVal;
}

int codeVersionTableMsg::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read purchaseType
    {
        purchaseType=(istream.readUI8());
        retVal+=1;
    }
    // read purchasePrice
    {
        purchasePrice=istream.readUI16();
        retVal+=2;
    }
    // read videoFormat
    retVal += videoFormat->read(istream);

    return retVal;
}

int codeVersionTableMsg::write(DataOutputStream& ostream) { // throws IOException


    {   /** fix dependent sizes for videoFormat **/
    }

    int retVal= 0;
    // write purchaseType
    ostream.writeUI8(purchaseType);
    retVal +=1;
    // write purchasePrice
    ostream.writeUI16(purchasePrice);
    retVal +=2;
    // write videoFormat
    if ( videoFormat != NULL ) {
        retVal +=videoFormat->write(ostream);
    }

    return retVal;
}

int codeVersionTableMsg::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("codeVersionTableMsg\n");
    dc.increaseIndent();
    int retVal= 0;
    // write purchaseType
    dc.indent();
    dc.getPs().print("purchaseType=");
    dc.getPs().println((long)purchaseType);
    // write purchasePrice
    dc.indent();
    dc.getPs().print("purchasePrice=");
    dc.getPs().println((long)purchasePrice);
    // write videoFormat
    if ( videoFormat != NULL ) {
        dc.indent();
        dc.getPs().println("videoFormat") ;
        retVal +=videoFormat->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for purchaseType
//ui8 codeVersionTableMsg::getPurchaseType()
//{
//return this->purchaseType;
//}

// Setter for purchaseType
//void codeVersionTableMsg::setPurchaseType(ui8 val)
//{
//this->purchaseType= val;
//}

// Getter for purchasePrice
//ui16 codeVersionTableMsg::getPurchasePrice()
//{
//return this->purchasePrice;
//}

// Setter for purchasePrice
//void codeVersionTableMsg::setPurchasePrice(ui16 val)
//{
//this->purchasePrice= val;
//}

// Getter for videoFormat
//org_himalay_ccif_DescriptionData__PTR_TYPE codeVersionTableMsg::getVideoFormat()
//{
//return this->videoFormat;
//}

// Setter for videoFormat
//void codeVersionTableMsg::setVideoFormat(org_himalay_ccif_DescriptionData__PTR_TYPE val)
//{
//this->videoFormat= val;
//}

int codeVersionTableMsg::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_codeVersionTableMsg__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::codeVersionTableMsg* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::codeVersionTableMsg* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code